thunder/core/interpreter.py

```diff
@@ -2588,6 +2593,70 @@ def impl(self, other):
         return _interpret_call(impl, self, other)
 
 
+def _collections_namedtuple_lookaside(
+    typename: str,
+    field_names: Iterable[str],
+    *,
+    rename: bool = False,
+    defaults: None | Iterable[Any] = None,
+    module: None | str = None,
+):
+    # Type checks {
+    assert wrapped_isinstance(typename, str)
+    assert wrapped_isinstance(field_names, Iterable)
+    assert wrapped_isinstance(rename, bool)
+    if defaults is not None:
+        assert wrapped_isinstance(defaults, Iterable)
+    if module is not None:
+        assert wrapped_isinstance(module, str)
+    # }
+
+    # Wrap defaults {
+    if not isinstance(rename, WrappedValue):
+        rename = wrap_const(rename)
+
+    if defaults is None:
+        defaults = wrap_const(defaults)
+
+    if module is None:
+        module = wrap_const(module)
+    # }
+
+    # Run opaque namedtuple {
+    @interpreter_needs_wrap
+    def create_namedtuple(typename: str, field_names: str, **kwargs):
+        namedtuple_type = collections.namedtuple(typename, field_names, **kwargs)
```

Comment:
If module is not in the kwargs it takes the module from the direct caller, i.e. it uses `thunder.core.interpreter`. Could you update providing the top stack frame's module if no module argument is given, please? 
```
In [1]:     from collections import namedtuple
   ...:     import thunder
   ...:     typename = "MyNamedTuple"
   ...:     field_names = ("a", "b", "c")
   ...: 
   ...:     # Test returnign just the type {
   ...:     def f():
   ...:         return namedtuple(typename, field_names)
   ...: 
   ...:     jf = thunder.jit(f)


In [2]: 

In [2]: jf()
Out[2]: thunder.core.interpreter.MyNamedTuple
```

Other than that, great work!

Reply:
Addressed!